chapter2.adoc

```diff
@@ -1,214 +1,207 @@
 
-The RISC-V Debug Specification defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the ebreak instruction. In Sdsec, those instructions such as mret, sret, uret, ecall, must either act as NOP or trigger an exception, stopping execution and setting `cmderr` to 3. Notably, these instructions retain their normal functionality during single stepping.
+[[maxdbgpriv]]
+[options="header"]
+.Determining maximum debug access privilege with mdbgen[i] and sdedbgalw
+|=========================================
+| mdbgen[i] | sdedbgalw | Maximum debug privilege allowed 
+| 1      | Don't care      | M                 
+| 0      | 1      | S(HS)             
+| 0      | 0      | None               
+|=========================================
+
+[NOTE]
+As the `prv` and `v` fields in dcsr are Write Any Read Legal (WARL) fields, the external debugger is able to read back the written value to determine the maximum debug privilege level.  
+ 
+==== Privilege Level Changing Instructions
+
+The RISC-V Debug Specification cite:[dbgspec] defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the ebreak instruction. In Sdsec, those instructions including mret, sret, uret, ecall, must either act as NOP or trigger an exception (stopping execution and setting `cmderr` to 3) in Program Buffer. Notably, these instructions retain their normal functionality during single stepping.
 
 === Trace
-The extension requires that trace availability from each hart is constrained by default. When Sdsec is supported, the optional sideband signal to trace encoder, sec_check[i] cite:[etrace], must be implemented for each hart i, and this signal must be reset to 1. The sec_check[i] signal is only cleared when trace is permitted by machine mode trace control or submachine mode trace control.
+When Sdsec is supported, the optional sideband signal to trace encoder, sec_check[i] cite:[etrace], must be implemented for each hart i, and this signal must be reset to 1 (disabled). The sec_check[i] signal is only cleared when trace is allowed by <<mtrcctl, M-mode trace control>> and/or <<sdtrcctl, supervisor domain trace control>>.
```

Comment:
I am not sure reset to 1 (disabled) statement is required here for the two reasons: 
- Upon reset, a hart’s privilege mode is set to M by Priv Spec. The following sections in this spec already defines the signal behaviour when a hart in M-mode. 
- Resetting to 1 potentially creates confusion. Some implementation potentially miss the very first instruction in trace

Reply:
I agree, will remove the reset value requirement.